File: 프로그래머스/1/147355. 크기가 작은 부분문자열/크기가 작은 부분문자열.java
class Solution {
    public int solution(String t, String p) {
        int answer = 0;
        int pLen = p.length();
        long pLong = Long.parseLong(p); // p를 long으로 변환

        // t에서 나올 수 있는 부분문자열 개수만큼 반복
        for (int i = 0; i <= t.length() - pLen; i++) {
            // p 길이만큼 substring
            String sub = t.substring(i, i + pLen);
            long subLong = Long.parseLong(sub); // 부분문자열을 long으로 변환
            
            // 크기 비교
            if (subLong <= pLong) {
                answer++;
            }
        }
        return answer;
    }
}

File: 프로그래머스/1/147355. 크기가 작은 부분문자열/크기가 작은 부분문자열.js
function solution(t, p) {
    let count = 0;
    const len = p.length;
    const numP = Number(p); // 또는 BigInt(p) p의 길이가 길 경우

    for (let i = 0; i <= t.length - len; i++) {
        // 부분 문자열 추출
        let sub = t.substring(i, i + len);
        // 숫자로 변환 후 비교
        if (Number(sub) <= numP) {
            count++;
        }
    }
    return count;
}
